components/micro-gateway-core/src/main/ballerina/src/gateway/filters/analytics_request_filter.bal

```diff
@@ -34,16 +34,27 @@ public type AnalyticsRequestFilter object {
     }
 
     public function filterResponse(http:Response response, http:FilterContext context) returns boolean {
-        if (isAnalyticsEnabled) {
+        printDebug(KEY_ANALYTICS_FILTER,"FilterResponse method in analytics_request_filter invoked.");
+        if (isAnalyticsEnabled || isgRPCAnalyticsEnabled) {
             boolean filterFailed = <boolean>context.attributes[FILTER_FAILED];
             if (context.attributes.hasKey(IS_THROTTLE_OUT)) {
                 boolean isThrottleOut = <boolean>context.attributes[IS_THROTTLE_OUT];
                 if (isThrottleOut) {
-                    ThrottleAnalyticsEventDTO | error throttleAnalyticsEventDTO = trap populateThrottleAnalyticsDTO(context);
+                    ThrottleAnalyticsEventDTO|error throttleAnalyticsEventDTO = trap populateThrottleAnalyticsDTO(context);
                     if (throttleAnalyticsEventDTO is ThrottleAnalyticsEventDTO) {
-                        EventDTO | error eventDTO = trap getEventFromThrottleData(throttleAnalyticsEventDTO);
+                        if (isgRPCAnalyticsEnabled) {
+                            // throttle stream gRPC Analytics
+                            AnalyticsStreamMessage message = createThrottleMessage(throttleAnalyticsEventDTO);
+                            printDebug(KEY_ANALYTICS_FILTER,"gRPC throttle stream message created.");
+                            future<()> publishedGRPCThrottleStream = start dataToAnalytics(message);
```

Comment:
The variable name doesn't look correct. Shall we make it like `dataPublishFuture` or something similar